invokeai/backend/model_manager/metadata/metadata_store.py

```diff
@@ -0,0 +1,216 @@
+# Copyright (c) 2023 Lincoln D. Stein and the InvokeAI Development Team
+"""
+SQL Storage for Model Metadata
+"""
+
+import sqlite3
+from typing import List, Optional, Set, Tuple
+
+from invokeai.app.services.shared.sqlite.sqlite_database import SqliteDatabase
+
+from .fetch import ModelMetadataFetchBase
+from .metadata_base import AnyModelRepoMetadata, UnknownMetadataException
+
+
+class ModelMetadataStore:
+    """Store, search and fetch model metadata retrieved from remote repositories."""
+
+    def __init__(self, db: SqliteDatabase):
+        """
+        Initialize a new object from preexisting sqlite3 connection and threading lock objects.
+
+        :param conn: sqlite3 connection object
+        :param lock: threading Lock object
+        """
+        super().__init__()
+        self._db = db
+        self._cursor = self._db.conn.cursor()
+        self._enable_foreign_key_constraints()
```

Comment:
This is handled here: https://github.com/invoke-ai/InvokeAI/blob/5d87578746bacffb5b5866ef543a67ab61869070/invokeai/app/services/shared/sqlite/sqlite_database.py#L47

Shouldn't need to do it again.

Reply:
Removed. This was carried over from the summer's model manager refactor version.